internal/api/pkce.go

```diff
@@ -0,0 +1,25 @@
+package api
+
+import (
+	"regexp"
+)
+
+const PKCE = "pkce"
+
+const MinCodeChallengeLength = 43
+const MaxCodeChallengeLength = 128
+
+var codeChallengePattern = regexp.MustCompile("^[a-zA-Z-._~0-9]+$")
```

Comment:
```suggestion
var codeChallengePattern = regexp.MustCompile("^[a-zA-Z._~0-9-]+$")
```

By putting `-` at the end it makes it less likely to accidentally refactor into a pattern that does not allow dashes.